src/ruby/ext/grpc/extconf.rb

```diff
@@ -24,10 +24,18 @@
 
 ENV['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
 
-ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
-ENV['CC'] = RbConfig::CONFIG['CC']
-ENV['CXX'] = RbConfig::CONFIG['CXX']
-ENV['LD'] = ENV['CC']
+if ENV['AR'].nil? || ENV['AR'].size == 0
+    ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
+end
+if ENV['CC'].nil? || ENV['CC'].size == 0
+    ENV['CC'] = RbConfig::CONFIG['CC']
+end
+if ENV['CXX'].nil? || ENV['CXX'].size == 0
+    ENV['CXX'] = RbConfig::CONFIG['CXX']
+end
+if ENV['LD'].nil? || ENV['LD'].size == 0
+    ENV['LD'] = ENV['CC']
```

Comment:
sorry wait, this should probably be `ENV['LD'] = ENV['CC']`, can you please update?

Sorry the delay in the merge, thanks for the ping

Reply:
type: should probably be `ENV['LD'] = ENV['LD']`

Reply:
I changed that, and while I agree that the linker should be able to be specified separately, I originally wrote it that way to respect the original code, just in case there was a an unknown reason for requiring the linker to have the same invocation as the compiler.  See:  https://github.com/grpc/grpc/blob/master/src/ruby/ext/grpc/extconf.rb#L30